WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt

```diff
@@ -0,0 +1,64 @@
+package org.wordpress.android.ui.jetpack.restore.usecases
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.FAILED
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.FINISHED
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.RUNNING
+import org.wordpress.android.fluxc.store.ActivityLogStore
+import org.wordpress.android.fluxc.store.ActivityLogStore.FetchRewindStatePayload
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Complete
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Failure.NetworkUnavailable
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Failure.RemoteRequestFailure
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Progress
+import org.wordpress.android.util.NetworkUtilsWrapper
+import javax.inject.Inject
+
+const val DELAY_MILLIS = 10000L
+
+class GetRestoreStatusUseCase @Inject constructor(
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val activityLogStore: ActivityLogStore
+) {
+    suspend fun getRestoreStatus(site: SiteModel, restoreId: Long) = flow {
+        // start off with a delay until "queued" status is implemented in RewindStatusModel
+        delay(DELAY_MILLIS)
+
+        while (true) {
+            if (!networkUtilsWrapper.isNetworkAvailable()) {
+                emit(NetworkUnavailable)
+                return@flow
+            }
+
+            val statusForSite = activityLogStore.getRewindStatusForSite(site)
+            val rewind = statusForSite?.rewind
```

Comment:
Minor (🔍): Consider removing `statusForSite` parameter since it is not used anywhere. What I mean is that these two lines can be merge: `val rewind = activityLogStore.getRewindStatusForSite(site)?.rewind`

Reply:
In commit c5b9f6d